Prototype optional parameters

acotty opened the pull request:
I need to eventually add support for enabling exception breakpoints.
But in order to do this first I have to be able to see what features the existing DAP debugger supports.
But in order to do this I need to add support for the optional parameters.

I have added optional parameters to the wxEVT_DAP_INITIALIZE_RESPONSE event / dap::InitializeResponse structure for the supported boolean options. I have added optional parameters to a bunch of other structures that I will need to add support in the dap.cpp ::From(const Json& json) function when I get to them.

Before I go too much further I would liket o find out if I am heading in the right direction with this or should I be doing something completely differently? If I am heading in the right direction then do you think the changes will be acceptable or not and if not can you let me know what I need to do toy make them acceptable. If I am doing it wrong can you point me in the right direction I should be going.

acotty committed:
Prototype optional parameters

eranif commented on dap/dap.hpp:
```diff
@@ -566,9 +569,211 @@ struct WXDLLIMPEXP_DAP InitializeRequest : public Request {
+using ChecksumAlgorithm = wxString;
+
+/// Information about the capabilities of a debug adapter.
+struct Capabilities {
```
all structs should be decorated with `WXDLLIMPEXP_DAP `

acotty replied:
Thanks. Will fix.

acotty replied:
Just pushed the changes to fix this.

eranif commented on dap/dap.hpp:
```diff
@@ -566,9 +569,211 @@ struct WXDLLIMPEXP_DAP InitializeRequest : public Request {
+
+    // Format to use for the rendered values in this column. TBD how the format
+    // strings looks like.
+    std::optional<wxString> format;
```
can't we use `wxAny` ? with value set to null to indicate "does not exist"

acotty replied:
I will check this out as I did not even look at wxAny. 

Thanks as this is exactly the feedback I was after.

acotty replied:
Changes pushed to change to use wxAny.

acotty committed:
Merge branch 'eranif:master' into Optional_Testing

acotty committed:
Converted to use wxWidgets wxAny
Added support for SetExceptionBreakpoints and SetExceptionBreakpointsResponse to enable IDE to change exceptions enabled/disabled if needed.
